### PR TITLE
Add Zoho People connector

### PR DESCRIPTION
**Pull request by kasthuriraajan**
# Description

Generate Zoho People connector using hand written OpenAPI Specification with the help of official API doc.

Fixes # https://github.com/ballerina-platform/ballerina-extended-library/issues/336

One line release note: 
- Add Zoho People connector

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
All operations are tested.

**Test Configuration**:
* Ballerina Version: 2201.0.0
* Operating System: 20.04
* Java SDK: 11

# Checklist:

### Changes to OpenAPI definition

- [X] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [X] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [X] Added license header at the top of each .bal file. 
- [X] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [X] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [X] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [X] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [X] Compiled the connector successfully with the targeted ballerina version. 
- [X] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by aneeshafedo — COMMENTED** [review]
All the fields in the records are optional. Haven't they specified optional fields in the API ?

**Comment by kasthuriraajan** [comment]
@aneeshafedo optional/required details is available in documentation for the request body fields. Relevant fields are marked as  required now. But  for the response body fields they don't mentioned the required fields explicitly in doc. So for response related records I guess we need to keep all fields as optional.

**Comment by aneeshafedo** [comment]
> @aneeshafedo optional/required details is available in documentation for the request body fields. Relevant fields are marked as required now. But for the response body fields they don't mentioned the required fields explicitly in doc. So for response related records I guess we need to keep all fields as optional.

Ack. Yes let's only make the fields that are mentioned as required in the API doc as required in OpenAPI.

File: openapi/zoho.people/Module.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+### Step 3: Invoke connector operation
+1. Now you can use the operations available within the connector. Note that they are in the form of remote operations.
+
+    Following is an example to get leave types of a specific employee. 
+
+    ```ballerina
+    public function main() {
+        userId = <USER_ID>
+        people:LeaveTypes|error leaveTypes = check zohoPeople->getLeaveTypes(userId);
```
Comment (Chuhaa) [inline openapi/zoho.people/Module.md:43]:
```suggestion
        people:LeaveTypes|error leaveTypes =  zohoPeople->getLeaveTypes(userId);
```

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**